sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpReceiver.cs

```diff
@@ -300,10 +300,19 @@ private static void CloseLink(RequestResponseAmqpLink link)
                     new TaskCompletionSource<IEnumerable<AmqpMessage>>(TaskCreationOptions
                         .RunContinuationsAsynchronously);
 
-                using var registration = cancellationToken.Register(static state =>
+                using var registration = cancellationToken.Register(state =>
                 {
                     var tcs = (TaskCompletionSource<IEnumerable<AmqpMessage>>)state;
                     tcs.TrySetCanceled();
+                    if (!_isSessionReceiver)
```

Comment:
Now things will work as we want except for the case of the session receiver, but I think we can support this in the future.